api/src/main/java/org/ccci/idm/user/Dn.java

```diff
@@ -112,6 +125,15 @@ public Component(@Nonnull final String type, @Nonnull final String value) {
         }
 
         @Override
+        public int compareTo(@Nonnull final Component o) {
```

Comment:
sort by `type`, then `value` case insensitively. I could be swayed to look at `value` before `type`.

that would change this ordering: `cn=Zebra ou=Aardvark` to `ou=Aardvark cn=Zebra`